crates/qpy/src/circuits.rs

```diff
@@ -0,0 +1,2174 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+// We use the following terminology:
+// 1. "Pack": To create a struct (from formats.rs) from the original data
+// 2. "Serialize": To create binary data (Bytes) from the original data
+// 3. "Write": To write to a file obj the serialization of the original data
+// Ideally, serialization is done by packing in a binrw-enhanced struct and using the
+// `write` method into a `Cursor` buffer, but there might be exceptions.
+
+use hashbrown::HashMap;
+use numpy::IntoPyArray;
+use numpy::ToPyArray;
+use pyo3::exceptions::{PyIOError, PyValueError};
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::types::IntoPyDict;
+use pyo3::types::{PyAny, PyBytes, PyDict, PyIterator, PyList, PyString, PyTuple, PyType};
+use pyo3::IntoPyObjectExt;
+
+use qiskit_circuit::bit::{
+    ClassicalRegister, PyClassicalRegister, PyClbit, QuantumRegister, Register, ShareableClbit,
+    ShareableQubit,
+};
+use qiskit_circuit::circuit_data::{CircuitData, CircuitStretchType, CircuitVarType};
+use qiskit_circuit::circuit_instruction::{CircuitInstruction, OperationFromPython};
+use qiskit_circuit::classical;
+use qiskit_circuit::imports;
+use qiskit_circuit::operations::{ArrayType, Operation, OperationRef, Param, StandardInstruction};
+use qiskit_circuit::packed_instruction::{PackedInstruction, PackedOperation};
+use qiskit_circuit::{Clbit, Qubit};
+
+use smallvec::SmallVec;
+use uuid::Uuid;
+
+use crate::annotations::AnnotationHandler;
+use crate::bytes::Bytes;
+use crate::consts::standard_gate_from_gate_class_name;
+use crate::formats;
+use crate::params::{pack_param, unpack_param};
+use crate::value::{
+    circuit_instruction_types, deserialize, deserialize_with_args, dumps_value,
+    expression_var_declaration, get_circuit_type_key, pack_generic_data, pack_standalone_var,
+    serialize, tags, unpack_generic_data, DumpedValue, ExpressionType, QPYReadData, QPYWriteData,
+};
+
+use crate::UnsupportedFeatureForVersion;
+
+const UNITARY_GATE_CLASS_NAME: &str = "UnitaryGate";
+// used when writing
+type CustomOperationsMap = HashMap<String, PackedOperation>;
+type CustomOperationsList = Vec<String>;
```

Comment:
I think we shouldn't make aliases like this. It's more confusing to have to look up what the "real" type is, and we're not buying much here in terms of brevity anyway since the types in question are pretty simple.

Reply:
Agreed. Removed these.